tests/flakyfunctional/events/31-dont-stall-succeeded.t

```diff
@@ -21,18 +21,13 @@ set_test_number 4
 
 install_suite "${TEST_NAME_BASE}" "${TEST_NAME_BASE}"
 run_ok "${TEST_NAME_BASE}-validate" cylc validate "${SUITE_NAME}"
+
 run_ok "${TEST_NAME_BASE}-run" cylc run "${SUITE_NAME}"
-TIMEOUT=$(($(date +%s) + 60))
-while (($(date +%s) < TIMEOUT)) && \
-    ! grep -q '\[t1\.1\] .* succeeded' "${SUITE_RUN_DIR}/log/suite/log"
-do
-    sleep 1
-done
+poll_suite_stopped
+
 grep_ok '\[t1\.1\] .*succeeded' "${SUITE_RUN_DIR}/log/suite/log"
 
-sleep 5
-cylc stop --max-polls=10 --interval=2 "${SUITE_NAME}" 1>'/dev/null' 2>&1
-run_fail "${TEST_NAME_BASE}-stall" \
+run_fail "${TEST_NAME_BASE}-not-stalled" \
     grep -q -F 'WARNING - suite stalled' "${SUITE_RUN_DIR}/log/suite/log"
```

Comment:
As far as I can tell this will still perform the objective of the test: "Test no stall when task pool has succeeded tasks only."